crates/proof-of-sql/src/base/scalar/scalar_ext.rs

```diff
@@ -30,9 +30,46 @@ pub trait ScalarExt: Scalar {
     fn into_u256_wrapping(self) -> U256 {
         U256::from(Into::<[u64; 4]>::into(self))
     }
+
+    /// Converts a slice of bytes into a `Scalar` by interpreting the bytes as
+    /// a little-endian number and taking it modulo the field’s modulus.
+    fn from_le_bytes_mod_order(bytes: &[u8]) -> Self;
```

Comment:
For this PR, we need to have a method that is
```rust
fn from_byte_slice_via_hash(bytes: &[u8]) -> Self {
    let mut hasher = tiny_keccak::Keccak::v256();
    hasher.update(bytes);
    let mut hashed_bytes = [0u8; 32];
    hasher.finalize(&mut hashed_bytes);
    // mask hashed_bytes with CHALLENGE_MASK
    // convert hashed_bytes to S
}
```

Alternatively, create the method and just copy the code from
```rust
impl<T: MontConfig<4>> From<&[u8]> for MontScalar<T>
```
but I don't know if that will be any easier.